Source/deviceinfo/DeviceInfo.cpp

```diff
@@ -611,106 +614,574 @@ class DeviceInfoLink : public WPEFramework::RPC::SmartInterfaceType<WPEFramework
 
 };
 
+class MemoryCachedDeviceInfo
+{
+public:
+    MemoryCachedDeviceInfo& operator=(const MemoryCachedDeviceInfo&) = delete;
+    MemoryCachedDeviceInfo(const MemoryCachedDeviceInfo&) = delete;
+    static MemoryCachedDeviceInfo& Instance()
+    {
+        return WPEFramework::Core::SingletonType<MemoryCachedDeviceInfo>::Instance();
+    }
+private:
+    MemoryCachedDeviceInfo():
+        _hdr_atmos_cec(0),
+        _outputResolution(),
+        _audioOutput(),
+        _videoOutput(),
+        _deviceInfoLink(DeviceInfoLink::Instance())
+    {
+        _architecture.Clear();
+        _chipsetName.Clear();
+        _modelName.Clear();
+        _modelYear.Clear();
+        _systemIntegraterName.Clear();
+        _friendlyName.Clear();
+        _platformName.Clear();
+        _firmwareVersion.Clear();
+        _idStr.Clear();
+        _supportedHdcp.Clear();
+        _maxOutputResolution.Clear();
+    }
+
+    ~MemoryCachedDeviceInfo()
+    {
+    }
+
+public:
+    uint32_t Deviceinfo_architecure(char buffer[], uint8_t* length)
+    {
+        uint32_t result = Core::ERROR_NONE;
+        if(_architecture.IsSet())
```

Comment:
As a coding guideline we always compare with something for readability. So in cases like this where you do not compare with something else specifically, we compare with == false or == true.
So in short this should be changed to if(_architecture.IsSet() == true).
Please also check for other occurrences in this file